### PR TITLE
Dev stack

### PR DESCRIPTION
**Pull request by romainGuiet**
Make LaRoMe work with a stack.

### CODE REVIEW

File: src/main/java/ch/epfl/biop/ij2command/Labels2Rois.java [hunk 1]
```diff
@@ -2,9 +2,11 @@
 
 import ij.ImagePlus ;
 import ij.IJ;
+import ij.gui.WaitForUserDialog;
```
Comment (NicoKiaru) [inline src/main/java/ch/epfl/biop/ij2command/Labels2Rois.java]: Can be removed

File: src/main/java/ch/epfl/biop/ij2command/Labels2Rois.java [hunk 2]
```diff
@@ -31,20 +33,49 @@
...
+
+        int nChannels   = dimensions[2];
+        int nSlices     = dimensions[3];
+        int nFrames     = dimensions[4];
+
+        if ( ((nChannels>1)&&(nSlices>1)) || ((nChannels>1)&&(nFrames>1)) || ((nSlices>1)&&(nFrames>1))){
+            System.out.println(""+imp.getTitle()+" is a hyperstack (multi c , z or t), please prepare a stack (single c, either z-stack or t-stack) from it.");
+            return;
+        } else if ((nChannels>1)||(nSlices>1)||(nFrames>1)){
+            System.out.println(""+imp.getTitle()+" is a stack of size"+copy_imp.getImageStackSize() );
```
Comment (NicoKiaru) [inline src/main/java/ch/epfl/biop/ij2command/Labels2Rois.java]: To comment

File: src/main/java/ch/epfl/biop/ij2command/Labels2Rois.java [hunk 3]
```diff
@@ -31,20 +33,49 @@
...
+            for (int i = 0; i < copy_imp.getImageStackSize(); i++) {
+                copy_imp.setPosition(i+1);
+                L2R( copy_imp );
+            }
+        } else {
+            System.out.println(""+imp.getTitle()+" is a single image");
```
Comment (NicoKiaru) [inline src/main/java/ch/epfl/biop/ij2command/Labels2Rois.java]: To comment

File: src/main/java/ch/epfl/biop/ij2command/Labels2Rois.java [hunk 4]
```diff
@@ -31,20 +33,49 @@
...
-        Wand wand = new Wand( imp.getProcessor() );
+        // get the imp dimension (width, height, nChannels, nSlices, nFrames)
+        int[] dimensions = copy_imp.getDimensions() ;
+
+        int nChannels   = dimensions[2];
+        int nSlices     = dimensions[3];
+        int nFrames     = dimensions[4];
+
+        if ( ((nChannels>1)&&(nSlices>1)) || ((nChannels>1)&&(nFrames>1)) || ((nSlices>1)&&(nFrames>1))){
+            System.out.println(""+imp.getTitle()+" is a hyperstack (multi c , z or t), please prepare a stack (single c, either z-stack or t-stack) from it.");
```
Comment (NicoKiaru) [inline src/main/java/ch/epfl/biop/ij2command/Labels2Rois.java]: Maybe use System.err

File: src/main/java/ch/epfl/biop/ij2command/Labels2Rois.java [hunk 5]
```diff
@@ -68,22 +97,25 @@ public void run() {
...
                     wand.autoOutline( x_coord, y_coord );
+
                     // if there is a region , then it has npoints
                     if ( wand.npoints > 0 ) {
                         // get the Polygon, fill with 0 and add to the manager
                         Roi roi = new PolygonRoi(wand.xpoints, wand.ypoints, wand.npoints, Roi.TRACED_ROI);
-                        ip.setRoi( roi );
-                        ip.fill();
+                        roi.setPosition( imp.getSlice() );
+                        ip.setRoi(roi);
```
Comment (NicoKiaru) [inline src/main/java/ch/epfl/biop/ij2command/Labels2Rois.java]: This line is probably not necessary

File: src/main/java/ch/epfl/biop/ij2command/Rois2Labels.java [hunk 6]
```diff
@@ -37,26 +40,70 @@
     @Override
     public void run() {
 
-        // reset RoiManager
+        // RoiManager
         RoiManager rm = RoiManager.getRoiManager();
```
Comment (NicoKiaru) [inline src/main/java/ch/epfl/biop/ij2command/Rois2Labels.java]: Why not as as input Parameter this time ?

File: src/main/java/ch/epfl/biop/ij2command/Rois2Labels.java [hunk 7]
```diff
@@ -37,26 +40,70 @@
...
+        int nSlices     = dimensions[3];
+        int nFrames     = dimensions[4];
+
+        ImagePlus label_imp;
+
+        if ( ((nChannels>1)&&(nSlices>1)) || ((nChannels>1)&&(nFrames>1)) || ((nSlices>1)&&(nFrames>1))){
+            System.out.println(""+imp.getTitle()+" is a hyperstack (multi c , z or t), please prepare a stack (single c, either z-stack or t-stack) from it.");
+            return;
+        } else if ((nChannels>1)||(nSlices>1)||(nFrames>1)) System.out.println(""+imp.getTitle()+" is a stack" );
+        else System.out.println(""+imp.getTitle()+" is a single image");
```
Comment (NicoKiaru) [inline src/main/java/ch/epfl/biop/ij2command/Rois2Labels.java]: COmment line System.out ?

File: src/main/java/ch/epfl/biop/ij2command/Rois2Labels.java [hunk 8]
```diff
@@ -37,26 +40,70 @@
...
-            ip.setValue(i+1);
-            ip.fill( rois[i] );
+        int nChannels   = dimensions[2];
+        int nSlices     = dimensions[3];
+        int nFrames     = dimensions[4];
+
+        ImagePlus label_imp;
+
+        if ( ((nChannels>1)&&(nSlices>1)) || ((nChannels>1)&&(nFrames>1)) || ((nSlices>1)&&(nFrames>1))){
+            System.out.println(""+imp.getTitle()+" is a hyperstack (multi c , z or t), please prepare a stack (single c, either z-stack or t-stack) from it.");
```
Comment (NicoKiaru) [inline src/main/java/ch/epfl/biop/ij2command/Rois2Labels.java]: System.err ?

**Review by NicoKiaru — CHANGES_REQUESTED** [review]
Few things to checks, please see comments

File: src/main/java/ch/epfl/biop/ij2command/Labels2Rois.java (hunk 1 above)
Comment (romainGuiet) [reply src/main/java/ch/epfl/biop/ij2command/Labels2Rois.java]: done

File: src/main/java/ch/epfl/biop/ij2command/Rois2Labels.java (hunk 6 above)
Comment (romainGuiet) [reply src/main/java/ch/epfl/biop/ij2command/Rois2Labels.java]: it is , forgot to remove this lien

File: src/main/java/ch/epfl/biop/ij2command/Labels2Rois.java (hunk 4 above)
Comment (romainGuiet) [reply src/main/java/ch/epfl/biop/ij2command/Labels2Rois.java]: done

File: src/main/java/ch/epfl/biop/ij2command/Labels2Rois.java (hunk 2 above)
Comment (romainGuiet) [reply src/main/java/ch/epfl/biop/ij2command/Labels2Rois.java]: done

File: src/main/java/ch/epfl/biop/ij2command/Labels2Rois.java (hunk 3 above)
Comment (romainGuiet) [reply src/main/java/ch/epfl/biop/ij2command/Labels2Rois.java]: done

File: src/main/java/ch/epfl/biop/ij2command/Labels2Rois.java (hunk 5 above)
Comment (romainGuiet) [reply src/main/java/ch/epfl/biop/ij2command/Labels2Rois.java]: indeed! removed!

File: src/main/java/ch/epfl/biop/ij2command/Rois2Labels.java (hunk 8 above)
Comment (romainGuiet) [reply src/main/java/ch/epfl/biop/ij2command/Rois2Labels.java]: done

File: src/main/java/ch/epfl/biop/ij2command/Rois2Labels.java (hunk 7 above)
Comment (romainGuiet) [reply src/main/java/ch/epfl/biop/ij2command/Rois2Labels.java]: removed

**Review by NicoKiaru — APPROVED** [review]

**Comment by NicoKiaru** [comment]
All good! For me, you can merge

**Pull request merged**